torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -127,6 +130,48 @@ RegisterOperators reg(
            };
          },
          aliasAnalysisSpecialCase()),
+     Operator(
+         prim::AllocateStorage, /*size, device*/
+         [](const Node* node) -> Operation {
+           size_t total_size = node->i(attr::total_size);
+           auto device = static_cast<DeviceType>(node->i(attr::device));
+           return [total_size, device](Stack* stack) {
+             at::DataPtr buffer =
+                 MemoryPlanner::allocate_buffer(total_size, device);
+             auto storage = c10::Storage(
+                 c10::Storage::use_byte_size_t(),
+                 total_size,
+                 std::move(buffer),
+                 /*allocator=*/nullptr,
+                 /*resizable=*/false);
+             push(stack, std::move(storage));
+           };
+         },
+         aliasAnalysisSpecialCase()),
+     Operator(
+         prim::AllocateTensor, /*size, sizes, offset, strides, dtype, device*/
+         [](const Node* node) -> Operation {
+           size_t size = node->i(attr::size);
+           size_t offset = node->i(attr::offset);
+           auto strides = node->is(attr::stride);
+           auto sizes = node->is(attr::sizes);
+           at::ScalarType dtype =
+               static_cast<at::ScalarType>(node->i(attr::dtype));
+           auto device = static_cast<DeviceType>(node->i(attr::device));
+
+           return [offset, size, sizes, strides, dtype, device](Stack* stack) {
+             c10::Storage buffer;
+             pop(stack, buffer);
+
+             uint8_t* start = static_cast<uint8_t*>(buffer.data());
+             void* src = static_cast<void*>(start + offset);
+             at::Tensor sub_tensor = at::from_blob(
```

Comment:
I briefly discussed this with @hlu1 - namely the fact that out variants should be calling resize on output themselves in order to get the striding right. greping the code base I only found calls like `output.resize_({sizeD, osizeT, osizeH, osizeW});` so I wasn't aware some kind of "one size fits all" like `.resize_(0)` (or `.resize_({0})`) does the right thing. It's actually not clear to me what's happening in that snippet since I don't see a second resize (unless you're implying that `.add` has an implicit resize...

Reply:
What we want to have happen is:
- allocate tensor of a given size, at an offset in the slab of memory
- resize_(0) so that the output of the node will set the correct strides
- when that resize happens, still use the same offset in memory for the newly resized tensor 

In the example, `y` and `z` are sharing memory. We resize `y` to 0, then we use it as the out of the add. `y` has the correct channels-last striding that we want, and we also show that it is still using the memory offset that it was initialized to when it is resized, because `y.sum() == z.sum()`. When we pass a tensor of the correct size (not resized to 0), we see that the channels-last memory is not preserved, striding is taken from the input.

Reply:
I think this is still relevant - but we can leave for a follow up. Maybe add comment of TODO ?